light/store/db/db.go

```diff
@@ -238,37 +243,30 @@ func (s *dbs) Prune(size uint16) error {
 	b := s.db.NewBatch()
 	defer b.Close()
 
-	pruned := 0
 	for itr.Valid() && numToPrune > 0 {
-		key := itr.Key()
-		height, err := s.decodeLbKey(key)
-		if err != nil {
-			return err
-		}
-		if err = b.Delete(s.lbKey(height)); err != nil {
-			return err
+		if err = b.Delete(itr.Key()); err != nil {
+			return
 		}
 		itr.Next()
 		numToPrune--
-		pruned++
 	}
 	if err = itr.Error(); err != nil {
-		return err
+		return
 	}
 
-	err = b.WriteSync()
-	if err != nil {
-		return err
-	}
-
-	// 3) Update size.
+	// 3) // update size
 	s.mtx.Lock()
-	defer s.mtx.Unlock()
+	s.size = size
+	s.mtx.Unlock()
 
-	s.size -= uint16(pruned)
+	if wErr := b.Set(s.sizeKey(), marshalSize(size)); wErr != nil {
+		err = fmt.Errorf("failed to persist size: %w", wErr)
+	}
 
-	if wErr := s.db.SetSync(s.sizeKey(), marshalSize(size)); wErr != nil {
-		return fmt.Errorf("failed to persist size: %w", wErr)
+	// 4) write batch deletion to disk
+	err = b.WriteSync()
```

Comment:
Do we want to apply the same logic to pruning the light client store as to the block and state store where after x amount of light blocks we flush to disk

Reply:
Depends on how much data there is to delete in a typical worst case. Batches are typically buffered in memory, so may OOM the node if they grow too large.

Reply:
A typical light block for a 100 validator network would be 15-17kB

Reply:
The batches don't store the blocks themselves, only the keys scheduled for deletion. So it would be average key size multiplied by worst-case number of keys. Some databases may also have limits on maximum batch sizes.

Reply:
I can't see this being a problem then